### PR TITLE
[flink][mysql-cdc] Introduce TypeMapping to specify data type mapping rules

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose

Main changes:
Introduce `TypeMapping`, move tinyint1ToBool to this class.
`MySqlSchema` builds data type;
`MySqlTableSchemaBuilder` builds data type;
`MySqlDebeziumJsonEventParser` handle schema changes;

### Tests

`MySqlCdcDataTypeOptionsITCase`
### API and Format

new: --type-mapping

### Documentation

CDC Ingestion

### CODE REVIEW

File: docs/content/how-to/cdc-ingestion.md [hunk 1]
```diff
@@ -147,7 +148,8 @@ To use this feature through `flink run`, run the following shell command.
     [--table-suffix <paimon-table-suffix>] \
     [--including-tables <mysql-table-name|name-regular-expr>] \
     [--excluding-tables <mysql-table-name|name-regular-expr>] \
-    [--mode <sync-mode>] \
```
Comment (JingsongLi) [inline docs/content/how-to/cdc-ingestion.md:150]: revert this.

**Review by JingsongLi — COMMENTED** [review]
Type mapping also needs to work with Kafka.

**Comment by yuzelin** [comment]
> Type mapping also needs to work with Kafka.

I'll open a new PR to do it.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**